lib/minfraud/components/email.rb

```diff
@@ -21,15 +25,38 @@ class Email < Base
       # @return [String, nil]
       attr_accessor :domain
 
+      # By default, the address will be sent in plain text. If this is set
+      # true, the address will instead be sent as an MD5 hash.
+      #
+      # @return [Boolean, nil]
+      attr_accessor :hash_address
+
       # @param params [Hash] Hash of parameters. Each key/value should
       #   correspond to one of the available attributes.
       def initialize(params = {})
-        @address = params[:address]
-        @domain  = params[:domain]
+        @address      = params[:address]
+        @domain       = params[:domain]
+        @hash_address = params[:hash_address]
 
         validate
       end
 
+      # A JSON representation of Minfraud::Components::Email.
+      #
+      # @return [Hash]
+      def to_json(*_args)
+        json = super
+
+        if json.delete('hash_address') && json['address']
+          hash = hash_email_address(json['address'])
+
+          # We could consider clearing the key if !hash.
+          json['address'] = hash if hash
```

Comment:
I wonder if we should set the `domain` if not set. This is what the Java and .NET libraries do:

https://github.com/maxmind/minfraud-api-java/blob/683cdc4a0dc05c7b1d97f02ee4c9c9d53c6d7119/src/main/java/com/maxmind/minfraud/request/Email.java#L72-L76

Reply:
Yeah, good idea! Thank you.